internal/dwarf/frame/table.go

```diff
@@ -296,21 +312,30 @@ func (ctx *Context) Execute(instructions []byte) *InstructionContextIterator {
 	}
 }
 
-func executeDWARFInstruction(ctx *Context) {
+func executeDWARFInstruction(ctx *Context) error {
 	instruction, err := ctx.buf.ReadByte()
 	if err != nil {
 		panic("Could not read from instruction buffer")
 	}
 
 	if instruction == DW_CFA_nop {
-		return
+		return nil
 	}
 
-	fn := lookupFunc(instruction, ctx)
+	fn, err := lookupFunc(instruction, ctx)
 	fn(ctx)
+	if err != nil {
+		return fmt.Errorf(" DWARF CFA rule is not valid. This should never happen :%w", err)
+	}
+	return nil
 }
 
-func lookupFunc(instruction byte, ctx *Context) instruction {
+// getPid gets the Process ID as a string from the fullExecutablePath provided.
+func getPid(fullExecutablePath string) string {
+	return strings.Split(fullExecutablePath, "/")[2]
```

Comment:
this is likely to panic, we already get this information by looking at the path, so I wouldn't try to parse it out of the path again, a human is going to look at this error anyway